terraform/api_gateway.tf.json.template.py

```diff
@@ -124,11 +128,15 @@ def public_ip() -> str:
     """
     Return the public IPv4 address of the machine running this code.
     """
-    url = 'https://checkip.amazonaws.com'
+    url = furl('https://checkip.amazonaws.com')
     http = http_client(log)
-    response = http.request('GET', url)
+    response = http.request('GET', str(url))
     assert response.status == 200, R('Unexpected response', response)
     ip_address = response.data.decode().strip()
+    try:
+        ipaddress.IPv4Address(ip_address)
+    except ipaddress.AddressValueError as e:
+        raise ValueError('Not a valid IPv4 address', ip_address) from e
```

Comment:
There is no point in wrapping the original exception. Remove the `try` altogether.